src/main/java/com/Alvaeron/api/RPEngineAPI.java

```diff
@@ -45,83 +47,111 @@ public static RoleplayPlayer getRoleplayPlayer(UUID uuid) {
      */
     @SuppressWarnings("deprecation")
     public static RoleplayPlayer getRoleplayPlayer(String playerName) {
-        if (Bukkit.getPlayer(playerName) != null) {
-            return Engine.manager.getPlayer(Bukkit.getPlayer(playerName).getUniqueId());
-        } else if (Bukkit.getOfflinePlayer(playerName) != null) {
-            Player p = (Player) Bukkit.getOfflinePlayer(playerName);
-            if (Engine.manager.getPlayer(p.getUniqueId()) != null) {
-                return Engine.manager.getPlayer(p.getUniqueId());
-            } else {
-                Engine.mm.createRoleplayPlayer(p);
-                return Engine.manager.getPlayer(p.getUniqueId());
-            }
-        } else {
+        Player player = Bukkit.getPlayer(playerName);
+        if (player != null) {
+            return Engine.manager.getPlayer(player.getUniqueId());
+        }
+
+        OfflinePlayer p = Bukkit.getOfflinePlayer(playerName);
```

Comment:
This is deprecated and really shouldn't be used since name changes exist and a name could change between logins

Reply:
I agree it shouldn't be used, but I didn't want to break the method contracts *for now*

Reply:
Convert this topic to an issue